### PR TITLE
ci: add gke provisioning

### PR DESCRIPTION
**Pull request by yangchiu**
ci: add gke provisioning

For https://github.com/longhorn/longhorn/issues/4621
    
Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

File: manager/integration/deploy/test.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ rules:
   resources: ["customresourcedefinitions"]
   verbs: ["*"]
 - apiGroups: ["policy"]
-  resources: ["podsecuritypolicies"]
+  resources: ["podsecuritypolicies", "poddisruptionbudgets"]
```
Comment (innobead) [inline manager/integration/deploy/test.yaml:32]: Why we need `poddisruptionbudgets`? will our test operate PDB directly?

File: test_framework/Jenkinsfile [hunk 2]
```diff
@@ -124,8 +126,12 @@ node {
                                        --env TF_VAR_tf_workspace=${TF_VAR_tf_workspace} \
                                        --env TF_VAR_tf_selinux_mode=${SELINUX_MODE} \
                                        --env TF_VAR_registration_code=${REGISTRATION_CODE} \
+                                       --env TF_VAR_gcp_project=${GCP_PROJECT} \
+                                       --env TF_VAR_gcp_auth_file=/src/longhorn-tests/gcp_creds.json \
                                        ${imageName}
                 """
+
+                sh "docker cp ${GCP_CREDS_FILE} ${JOB_BASE_NAME}-${BUILD_NUMBER}:/src/longhorn-tests/gcp_creds.json"
```
Comment (innobead) [inline test_framework/Jenkinsfile:134]: This will only execute when `GCP_CREDS_FILE` exists.

File: test_framework/scripts/longhorn-setup.sh [hunk 3]
```diff
@@ -25,16 +25,22 @@ set_kubeconfig_envvar(){
...
+    elif [[ ${TF_VAR_k8s_distro_name} == "gke" ]]; then
+      gcloud container clusters get-credentials `terraform -chdir=${TF_VAR_tf_workspace}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${TF_VAR_k8s_distro_name} output -raw cluster_name` --zone `terraform -chdir=${TF_VAR_tf_workspace}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${TF_VAR_k8s_distro_name} output -raw cluster_zone` --project ${TF_VAR_gcp_project}
     else
-      export KUBECONFIG="${BASEDIR}/terraform/aws/${DISTRO}/k3s.yaml"
+      export KUBECONFIG="${BASEDIR}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${DISTRO}/k3s.yaml"
     fi
   elif [[ ${ARCH} == "arm64"  ]]; then
-    export KUBECONFIG="${BASEDIR}/terraform/aws/${DISTRO}/k3s.yaml"
+    if [[ ${TF_VAR_k8s_distro_name} == "gke" ]]; then
+      gcloud container clusters get-credentials `terraform -chdir=${TF_VAR_tf_workspace}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${TF_VAR_k8s_distro_name} output -raw cluster_name` --zone `terraform -chdir=${TF_VAR_tf_workspace}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${TF_VAR_k8s_distro_name} output -raw cluster_zone` --project ${TF_VAR_gcp_project}
```
Comment (innobead) [inline test_framework/scripts/longhorn-setup.sh:40]: note: `It takes a project and a zone as parameters, passed through by set defaults or flags. By default, credentials are written to HOME/.kube/config`

ref: https://cloud.google.com/sdk/gcloud/reference/container/clusters/get-credentials

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM. Just a few questoins needed to answer.

GKE provision looks simpler.

File: manager/integration/deploy/test.yaml (hunk 1 above)
Comment (yangchiu) [reply manager/integration/deploy/test.yaml:32]: This follows the CA test case PR: https://github.com/longhorn/longhorn-tests/pull/936/files#diff-deb01c8f2cd2d469bfeed0826ebac624022e4b668a4f7d1d2f04d4d9c22d5f82R32, not sure if it's necessary.

File: test_framework/Jenkinsfile (hunk 2 above)
Comment (yangchiu) [reply test_framework/Jenkinsfile:134]: Yes, `GCP_CREDS_FILE` is a secret stored in Jenkins, just like the AWS credentials. https://github.com/longhorn/longhorn-tests/pull/1312/files#diff-07a02be3672fe9e5e5da2e99ceb6f30f23b939d8a849521cdb250a007276e8edR33

File: manager/integration/deploy/test.yaml (hunk 1 above)
Comment (yangchiu) [reply manager/integration/deploy/test.yaml]: This doesn't show anymore, because I rebased this PR.

Comment (yangchiu) [reply manager/integration/deploy/test.yaml]: I've tried with or without `poddisruptionbudgets`, but can't tell the differences. @c3y1huang can you help to elaborate more? Thanks!

**Review by innobead — APPROVED** [review]
LGTM.

Feel the Jenkinsfile becomes a monster config..

**Pull request merged**